keystone_performance/load_test.py

```diff
@@ -222,47 +294,54 @@ def _print(self):
 
 
 class Request(object):
-    def __init__(self, agent, request_gatherer, args, on_complete=None):
+    def __init__(self, agent, request_gatherer, args, user_info, on_complete=None):
         self._agent = agent
         self._request_gatherer = request_gatherer
         self._args = args
+        self._user_info = user_info
         self._on_complete = on_complete
 
         self._request_no = 0
         self._done = False
 
-        if self._args.user_domain_id:
-            user_domain_info = {'id': self._args.user_domain_id}
-        else:
-            user_domain_info = {'name': self._args.user_domain_name}
-        if self._args.project_id:
-            project_info = {'id': self._args.project_id}
-        else:
-            project_info = {'name': self._args.project_name}
-        if self._args.project_domain_id:
-            project_info['domain'] = {'id': self._args.project_domain_id}
+    def start(self):
+        user_info = self._user_info.get_random_user_info()
+
+        project_info = None
+        if 'project_id' in user_info:
+            project_info = {'id': user_info['project_id']}
+        elif 'project_name' in user_info:
+            project_info = {'name': user_info['project_name']}
+        if 'project_domain_id' in user_info:
+            project_info['domain'] = {'id': user_info['project_domain_id']}
+        elif 'project_domain_name' in user_info:
+            project_info['domain'] = {'name': user_info['project_domain_name']}
+        if project_info:
+            scope = {'project': project_info}
         else:
-            project_info['domain'] = {'name': self._args.project_domain_name}
+            if 'domain_id' in user_info:
+                user_domain_info = {'id': user_info['domain_id']}
+            elif 'domain_name' in user_info:
+                user_domain_info = {'name': user_info['domain_name']}
+            scope = {'domain': user_domain_info}
+        
         auth_req_body = {
             'auth': {
                 'identity': {
                     'methods': ['password'],
                     'password': {
                         'user': {
-                            'name': self._args.username,
-                            'domain': user_domain_info,
-                            'password': self._args.password
+                            'name': user_info['username'],
+                            'domain': {'id': user_info['domain_id']},
+                            'password': 'testing'
```

Comment:
We cannot have passwords checked into the code. This needs to be passed in as a parameter.